compiler/rustc_hir_typeck/src/callee.rs

```diff
@@ -722,12 +731,57 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         err.emit()
     }
 
+    /// Enforces that things being called actually are callable
+    #[instrument(skip(self, arguments))]
+    pub(super) fn check_callable(
+        &self,
+        hir_id: hir::HirId,
+        span: Span,
+        value: Ty<'tcx>,
+        arguments: impl IntoIterator<Item = Ty<'tcx>>,
+    ) {
+        if value.references_error() {
+            return;
+        }
+
+        let context = self.tcx.hir().enclosing_body_owner(hir_id);
+        let kind = self.tcx.def_kind(context.to_def_id());
+        debug_assert_ne!(kind, DefKind::ConstParam);
+
+        let ty::FnDef(did, substs) = *value.kind() else {
```

Comment:
Why are we not also registering these predicates for other types?

Reply:
this is leftover cruft from the constness logic. We don't need it anymore now that we use the `Callable` trait. w